src/pages/ProductDetail/index.tsx

```diff
@@ -9,41 +9,58 @@ import { Product, Loading, ProductList } from '@/components';
 const ProductDetail = () => {
   const { id } = useParams<{ id: string }>();
 
+  // Use the useProductById hook for the product and status
+  const { product, isLoading: isProductLoading, status } = useProductById(id || '');
+
   // Use the useProducts hook for related products
-  const { product, isLoading: isProductLoading } = useProductById(id || '');
   const { products, isLoading: isRelatedProductsLoading, hasMore, onLoadMore } = useProducts(product?.category || '');
 
+  // Determine loading state
+  const isLoading = isProductLoading && status === undefined;
+
   return (
     <MainLayout>
-      <div>
-        <div className={styles.product}>
-          {isProductLoading ? (
-            <Loading classStyle={styleUtils.loading} />
-          ) : (
-            product && (
-              <Product
-                colors={product.colors || []}
-                description={product.description || ''}
-                images={product.images || []}
-                numberRating={product.numberRating || 0}
-                price={product.price || 0}
-                rate={product.rate || 0}
-                sizes={product.sizes || []}
-                stock={product.stock || 0}
-                title={product.title || ''}
-              />
-            )
-          )}
+      {isLoading ? (
+        <Loading classStyle={`${styleUtils.loading} ${styles.loading}`} />
+      ) : status === 404 ? (
```

Comment:
create variable for this `const isNotFound = status === STATUS.NOT_FOUND`

Reply:
Fixed